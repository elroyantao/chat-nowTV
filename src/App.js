import React, { Component } from 'react';
import PropTypes from 'prop-types'
import { connect } from 'react-redux';

import getChatLog from './service';

import './App.css';
import ChatRoom from './components/Chat/ChatRoom'

class App extends Component {
  static propTypes = {
    getChatLog: PropTypes.func
  }
  componentDidMount() {
    const { getChatLog } = this.props
    getChatLog()
  }
  render() {
    return (
      <div>
        <h1>Welcome to React.js chat room!</h1>
        <ChatRoom />
      </div>
    );
  }
}

const mapDispatchToProps = { getChatLog }

export default connect(
  null,
  mapDispatchToProps
)(App);
