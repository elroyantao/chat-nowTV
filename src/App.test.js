import testHelper from './helpers/test-helper'
import App from './App';

// it('should render without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App.WrappedComponent />, div);
// });

describe('<App />', () => {
  const renderCompoennt = testHelper(App.WrappedComponent)
  const props = {
    getChatLog: jest.fn()
  }
  beforeEach(() => props.getChatLog.mockClear())      
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderCompoennt(props).getTree())
        .toMatchSnapshot()
    })
  })
  describe('@lifecycle', () => {
    describe('componentDidMount', () => {
      it('should call getChatLog', () => {
        const { instance } = renderCompoennt(props)
        expect(instance.props.getChatLog).toHaveBeenCalledTimes(1)
      })
    })
  })
})