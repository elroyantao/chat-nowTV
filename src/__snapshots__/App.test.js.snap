exports[`<App /> @renders in default state 1`] = `
<div>
  <h1>
    Welcome to React.js chat room!
  </h1>
  <Connect(ChatRoom) />
</div>
`;
